src/crystal/system/darwin/process.cr

```diff
@@ -0,0 +1,18 @@
+require "c/mach-o/dyld"
+
+module System
+  # nodoc
+  class Process
+    def self.executable_path_impl
+      buf = GC.malloc_atomic(LibC::PATH_MAX).as(UInt8*)
+      size = LibC::PATH_MAX.to_u32
+
+      if LibC._NSGetExecutablePath(buf, pointerof(size)) == -1
+        buf = GC.malloc_atomic(size).as(UInt8*)
+        return nil if LibC._NSGetExecutablePath(buf, pointerof(size)) == -1
```

Comment:
No. The first call will report an error and update `size` to be the actual path length, when the default, arbitrary-sized, buffer is too small. We must make a *second* call with a larger buffer to get the path.